Signup: Unique copy for button text

SnGrimes opened the pull request:
This branch is associated with #25216.

Currently, the button copy for the Onboarding checklist items are the same. @alisterscott suggested changing the copy for each button to break up the repetition. 
The image below is just a sample of what the buttons look like with different text. 

Before 
![image](https://user-images.githubusercontent.com/14166723/40945149-09afdb6a-681d-11e8-9847-b4fed3221633.png)

**After**
![wp-calypso_onboard_checklist_edits](https://user-images.githubusercontent.com/14166723/40944886-07bbfb64-681c-11e8-96c3-dc5a29fb8a51.PNG)
 
I had to use the React add on in dev tools to change the state because I'm currently receiving 304 or 404 messages and the changes I've made are not displaying in the browser even on rebuild.
My plan is to add `buttonText` to the props for the onboardingChecklist.js and remove `{ buttonText = translate( 'Do it!' ) } = this.props` from checklist-task.jsx.

alisterscott commented on client/blocks/checklist/checklist-task.jsx:
```diff
@@ -56,8 +56,9 @@ export class ChecklistTask extends PureComponent {
+			buttonText,
 		} = this.props;
-		const { buttonText = translate( 'Do it!' ) } = this.props;
+		//{ buttonText = translate( 'Do it!' ) } = this.props;
```
Could you please remove this commented line?

SnGrimes replied:
Awesome! I will remove the commented line but I do have a question about ```translate()```.  This text probably still needs to be translated, so would I also need to fit ```translate()``` in there somewhere? I was reading the i18n and it mentioned using args in order to use variables.

jblz replied:
Ideally, all these should be wrapped in a `translate` call, yes.

You can add that function as an **external** dependency like so:
`import { translate } from 'i18n-calypso';`

It does make it a more complex from a design perspective because different languages will have different string lengths.

I'd look into:
* Specifying reasonable `max-width` values at appropriate breakpoints somewhere in [this file](https://github.com/Automattic/wp-calypso/blob/3ea0e1f44871d84f18a654a9a4f81ad88bb174fa/client/blocks/checklist/style.scss)
* Adding the `buttonText` as the value of a `title` attribute on the `Button` component [here](https://github.com/Automattic/wp-calypso/blob/3ea0e1f44871d84f18a654a9a4f81ad88bb174fa/client/blocks/checklist/checklist-task.jsx#L85) so that strings that run long are still accessible via mouse-over

SnGrimes commented:
@alisterscott I removed the commented line and added translate back in for the buttonText. I also added buttonText to the title attribute as suggested by @jblz.

sirreal commented on client/blocks/checklist/checklist-task.jsx:
```diff
@@ -56,8 +56,8 @@ export class ChecklistTask extends PureComponent {
 			duration,
 			title,
 			translate,
+			buttonText,
```
The existing implementation has a fallback value, it would be a good idea to keep that so we don't unexpectedly break anything 🙂 

We can just restore the previous default right here:

```js
buttonText = translate( 'Do it!' ),
```

sirreal commented on client/blocks/checklist/checklist-task.jsx:
```diff
@@ -86,8 +86,11 @@ export class ChecklistTask extends PureComponent {
-						{ hasActionlink ? completedButtonText : buttonText }
+						{ hasActionlink
+							? completedButtonText
+							: translate( '%(buttonText)s!', { args: { buttonText: buttonText } } ) }
```
If we apply a translation here, a translator will see `[ someVariable ]!`. The can only see an exclamation point to translate, the rest of the text is hidden from them 😮 🙈 

It's a good idea to apply the translation to the whole string, so we should drop the changes on this line and apply the translations elsewhere 👍

sirreal commented on client/my-sites/checklist/onboardingChecklist.js:
```diff
@@ -31,6 +31,7 @@ const tasks = {
 		url: '/me',
 		image: '/calypso/images/stats/tasks/upload-profile-picture.svg',
 		tour: 'checklistUserAvatar',
+		buttonText: 'Add Profile',
```
Each of these texts are what we translate, ideally. Including punctuation if necessary.

In order to do that, we'll need to add the import in the `External dependencies` block:

```js
import { translate } from 'i18n-calypso';
```

Then wrap the texts as so: `translate( 'Add Profile' )`.

michelleweber commented:
(Before commenting on the new stuff, I'll say:  I actually like the repetitive buttons, which, for me, create a feeling of "each of these tasks in similarly small and manageable, and you can run through them, no problem" and keeps the focus on the main text. And "Do it" is a complete sentence, which is nice from a tone POV.)

On the new version:
- I'd either lose the exclamation points with this new button text (these aren't actual sentences, so punctuation feels a little off), or I'd made these all actual sentences (my vote): Name your site, Upload an icon, Create a tagline, Add a photo, Tweak your page, Publish a post.
- Make sure capitalization is consistent; the second word should be lowercase across the board.
- You've got three "Add"s in a row, so I'd mix those up a little. Upload an icon, Create a tagline, Add a photo.

sirreal commented:
@michelleweber Do you think these changes are worthwhile? (#25216)

michelleweber commented:
I'd want to know: is there a reason for changing it? Like, is there a drop-off as people move through the checklist, and you suspect this has something to do with it? In that case, I'd A/B it, or change it and the look at things in 2 weeks to see if it makes any difference.

alisterscott commented:
@michelleweber

From: https://www.useronboard.com/how-wordpress-onboards-new-users/?slide=57

> The button copy is a bit repetitive when it's just the same phrase over and over

Matt M would like us to address issues raised in that feedback

michelleweber commented:
Fair enough; in that case, see my notes on the update above -- if anything's unclear and/or you want another look after you make the changes, just ping!

SnGrimes commented:
Alright, well I'm waiting to get my Linux laptop back from the repair shop. I can't really use wp-calypso on my windows machine without code changes taking 3 times longer than it should.

alisterscott commented:
I might close this one as it's out of date with master - feel free to open a new PR if you like 👍
